src/indexer/manager.ts

```diff
@@ -162,29 +167,31 @@ export class IndexManager extends Component {
 
     if (newIndexer) {
       logger.debug(
-        "[file:%s] using indexer %s for file",
-        indexKey,
+        "[indexer:%s] [file:%s] attempting to index",
         newIndexer.id,
+        indexKey,
       );
 
-      let result: ReturnType<Indexer["onChanged"]>;
+      let result: ReturnType<Indexer["onChanged"]> | undefined = undefined;
       try {
-        result = newIndexer.onChanged(file.path, cache);
+        result = newIndexer.onChanged(file, cache);
       } catch (error) {
-        result = "error";
         logger.error(
           "[indexer:%s] [file:%s] unexpected error or result while indexing %o",
           newIndexer.id,
           indexKey,
           error,
         );
       }
+
       switch (result) {
         case "indexed":
+        case "error":
           logger.debug(
-            "[indexer:%s] [file:%s] indexed",
+            "[indexer:%s] [file:%s] %s",
             newIndexer.id,
             indexKey,
+            result == "error" ? "handled error" : "indexed",
```

Comment:
you're no longer setting `result` to `"error"` above. Do you want to do a different check?

Reply:
Oh, the indexer can return `error` if it has a "handled" error. That happens in cases where, e.g., there is a zod parsing error which gets recorded in the tree. The above error path is when something truly unexpected happens and the indexer itself errors out. In that case, we already get a logged message in the catch block, so we don't need to do a second log. This is all a little convoluted and I may merge several of these classes at some point, which might make things simpler.